master/internal/api_tasks.go

```diff
@@ -219,7 +219,27 @@ func (a *apiServer) GetAllocation(
 	ctx context.Context,
 	req *apiv1.GetAllocationRequest,
 ) (*apiv1.GetAllocationResponse, error) {
-	return nil, grpcutil.UnimplementedError
+	if req.AllocationId == "" {
+		return nil, status.Error(codes.InvalidArgument, "allocation ID missing")
+	}
+
+	resp := model.Allocation{}
```

Comment:
nit: https://github.com/uber-go/guide/blob/master/style.md#use-var-for-zero-value-structs